common-libs/common/src/services/DidDocumentService/index.ts

```diff
@@ -1,46 +1,34 @@
 import KeysService from '../KeysService'
 import JoloDidDocument from './JoloDidDocument'
 import ElemDidDocument from './ElemDidDocument'
+import ElemAnchoredDidDocument from './ElemAnchoredDidDocument'
 import { parse } from 'did-resolver'
 import { LocalKeyVault } from './LocalKeyVault'
+import { DidResolver } from '../../shared/DidResolver'
 
 export { default as ElemDidDocument } from './ElemDidDocument'
 export { KeyVault } from './KeyVault'
 export { LocalKeyVault } from './LocalKeyVault'
 
 export default class DidDocumentService {
-  constructor(keysService: KeysService) {
-    const { didMethod } = keysService.decryptSeed()
-
-    let didDocumentService
-    switch (didMethod) {
-      case 'jolo':
-        didDocumentService = new JoloDidDocument(keysService)
-        break
-      case 'elem':
-        didDocumentService = new ElemDidDocument(new LocalKeyVault(keysService))
-        break
-    }
-
-    return didDocumentService
-  }
-
-  getMyDid() {
-    return 'did:...'
+  /**
+   * @deprecated use DidDocumentService.createDidDocumentService instead
+   */
+  constructor(keysService: KeysService, didResolver?: DidResolver) {
+    return DidDocumentService.createDidDocumentService(keysService, didResolver)
   }
 
-  getKeyId(did: string = null) {
-    if (!did) {
-      did = this.getMyDid()
-    }
-
-    const signingKey = 'primary'
-
-    return `${did}#${signingKey}`
-  }
+  static createDidDocumentService(
+    keysService: KeysService,
+    didResolver?: DidResolver,
```

Comment:
If `didResolver` is optional here, shouldn't it throw an exception when `didMethod` is `'elem-anchored'` (instead of failing down the road)?

Reply:
OK, I see now that we shouldn't, so probably some comment is needed here to clarify that `didResolver` is only needed when we are trying to `buildDidDocument`.

Or maybe the better idea would be to remove this parameter from constructor entirely, and make `didResolver` a parameter for `buildDidDocument` method. That way it will reduce the opportunity to make bugs by calling `createDidDocumentService` without a second argument (how it is already done in some places) and then calling `buildDidDocument` on a result.

Reply:
hmm, if we will pass it as an argument, it means we need to **update** interface of the method or **detect** which type of DidDocumentService is used - not the best way from my side. 
I expect it won't make any issues in future if we will follow good test coverage, even more.